src/proto_tcp.c

```diff
@@ -583,31 +679,33 @@ int tcp_bind_listener(struct listener *listener, char *errmsg, int errlen)
 	}
 
 #if defined(TCP_MAXSEG)
-	if (listener->bind_conf->maxseg > 0) {
-		if (setsockopt(fd, IPPROTO_TCP, TCP_MAXSEG,
-			       &listener->bind_conf->maxseg, sizeof(listener->bind_conf->maxseg)) == -1) {
-			chunk_appendf(msg, "%scannot set MSS to %d, (%s)", msg->data ? ", " : "", listener->bind_conf->maxseg,
-				      strerror(errno));
-			err |= ERR_WARN;
-		}
-	} else {
-		/* we may want to try to restore the default MSS if the socket was inherited */
-		int tmpmaxseg = -1;
-		int defaultmss;
-		socklen_t len = sizeof(tmpmaxseg);
-
-		if (listener->rx.addr.ss_family == AF_INET)
-			defaultmss = sock_inet_tcp_maxseg_default;
-		else
-			defaultmss = sock_inet6_tcp_maxseg_default;
-
-		getsockopt(fd, IPPROTO_TCP, TCP_MAXSEG, &tmpmaxseg, &len);
-		if (defaultmss > 0 &&
-		    tmpmaxseg != defaultmss &&
-		    setsockopt(fd, IPPROTO_TCP, TCP_MAXSEG, &defaultmss, sizeof(defaultmss)) == -1) {
-			chunk_appendf(msg, "%scannot set MSS to %d, (%s)", msg->data ? ", " : "", defaultmss,
-				      strerror(errno));
-			err |= ERR_WARN;
+	if (listener->rx.proto->sock_prot != IPPROTO_MPTCP){
```

Comment:
you cannot do that: `TCP_MAXSEG` is currently not supported by MPTCP, but it will probably be later on. Hard to tell without trying.

Here below, there are two cases (from what I understand):
- the user has set a custom value → we need to try, and warn if it is not OK
- the system found a MSS ealier, it will try to apply it, see where `sock_inet_tcp_maxseg_default` and `sock_inet6_tcp_maxseg_default` are set → that's where you can probably do something: I guess they use a TCP socket. You can do the same check for MPTCP, and here, if the proto is MPTCP, set the default from MPTCP.

Maybe best to do that in a separated commit I think

(also, the code style is not OK: space between `){`)

Reply:
Added 2 variables sock_inet(6)_mptcp_maxseg_defaul, that will store the defaults for MPTCP. Note however that until the support for TCP_MAXSEG is added, they will always hold -1.